src/esp/assets/ResourceManager.cpp

```diff
@@ -2849,10 +2901,53 @@ void ResourceManager::addComponent(
                  child,          // mesh transform node
                  visNodeCache,   // a vector of scene nodes, the visNodeCache
                  computeAbsoluteAABBs,  // compute absolute aabbs
-                 staticDrawableInfo);   // a vector of static drawable info
+                 staticDrawableInfo,    // a vector of static drawable info
+                 skinData);             // instance skinning data
   }
 }  // addComponent
 
+void ResourceManager::mapSkinnedModelToArticulatedObject(
+    const MeshTransformNode& meshTransformNode,
+    const std::shared_ptr<physics::ArticulatedObject>& rig,
+    const std::shared_ptr<gfx::InstanceSkinData>& skinData) {
+  // Find skin joint ID that matches the node
+  const auto& gfxBoneName = meshTransformNode.name;
+  const auto& boneNameJointIdMap = skinData->skinData->boneNameJointIdMap;
+  const auto jointIt = boneNameJointIdMap.find(gfxBoneName);
+  if (jointIt != boneNameJointIdMap.end()) {
+    int jointId = jointIt->second;
+
+    // Find articulated object link ID that matches the node
+    const auto& linkIds = rig->getLinkIdsWithBase();
```

Comment:
Doesn't `getLinkIdsWithBase` return a vector by value? Are you making a reference to a temporary?!?